create_shards.c

```diff
@@ -97,6 +106,35 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 		ereport(ERROR, (errmsg("unsupported partition method: %c", partitionMethod)));
 	}
 
+	partitionColumnVar = ColumnNameToColumn(distributedTableId, partitionColumnName);
+	partitionColumnTypeId = partitionColumnVar->vartype;
+	partitionColumnOpClassId = GetDefaultOpClass(partitionColumnTypeId, BTREE_AM_OID);
+
+	/* if there is no operator class for the partition column, error out */
+	if (partitionColumnOpClassId == InvalidOid)
+	{
+		ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
+						errmsg("cannot distribute relation: \"%s\"", tableName),
+						errdetail("Column \"%s\" is not eligible to be the distribution "
+								  "column.", partitionColumnName),
+						errhint("Try another column as the distribution column.")));
+	}
+
+	lessEqualOperatorId = GetOperatorByType(partitionColumnTypeId, BTREE_AM_OID,
+											BTLessEqualStrategyNumber);
+	greaterEqualOperatorId =  GetOperatorByType(partitionColumnTypeId, BTREE_AM_OID,
+												BTGreaterEqualStrategyNumber);
+
+	/* if partition column does not support <= and >= operators, error out*/
+	if (lessEqualOperatorId == InvalidOid ||  greaterEqualOperatorId == InvalidOid)
```

Comment:
I could not find any type, which has non-InvalidOid partitionColumnOpClassId, with invalid equality operators.
But, to be on the safe side, I added this extra control.